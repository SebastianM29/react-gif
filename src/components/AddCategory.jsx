import React, { useState } from 'react'
import  PropTypes  from "prop-types";


//cada componente puede tener su propio estado
export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('')
    
    const onInputChange = (e) => {
      
      setInputValue(e.target.value)
      
    } 
    
    const capturandoValor = (e) => {
     e.preventDefault();
     
     if (inputValue.trim().length <= 1) {
      return
     }
  

     //console.log(inputValue)
     //setCategories((algos)=> {
     // console.log('este',algos)
      // return [...algos,inputValue]
     //})

     setInputValue('')
     onNewCategory(inputValue.trim())
     
    }

   

    return (
      <form onSubmit={(e) => capturandoValor(e)} aria-label='form'>
        <input 
          type="text"
          placeholder='Buscar imagenes'
          value={inputValue}
          //se puede mandar asi ya que sse recive el evento y se envia el evento
          //onChange={ (e) => { onInputChange}}
          //onchange cambia el inputValue
          onChange={ (e) => { onInputChange(e)}}
          />
       </form>
  )
}


AddCategory.propTypes = {
  onNewCategory : PropTypes.func.isRequired
}