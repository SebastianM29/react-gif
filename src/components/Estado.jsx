

export const Estado = ({status}) => {
    console.log('este es el status',status)
    let mensaje
    if (status) {
      mensaje = 'cargando'        
    }
  return (
    <div>{mensaje}</div>
  )
}
