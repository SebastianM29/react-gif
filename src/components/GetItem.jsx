import PropTypes from 'prop-types'

export const GetItem = ({title,image,id}) => {
  return (
    <div className="card">
        <img src={image} alt={title} />
        <p>{title}</p>
      
    </div>

  )
}

GetItem.propTypes = {
  title : PropTypes.string.isRequired,
  id: PropTypes.string.isRequired
}