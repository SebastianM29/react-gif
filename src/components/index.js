

export * from "./AddCategory";
export * from "./Estado";
export * from "./GetItem";
export * from "./Gifexp";