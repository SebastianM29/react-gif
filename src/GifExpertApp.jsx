
import { useState } from "react"
import { AddCategory,Gifexp } from "./components";



export const GifExpertApp = () => {
  const [categories, setCategories] = useState([])
  console.log(categories)

  const onAddCategory = (valorObtenido) => {
   console.log(valorObtenido)
   const valorObtenidoMayuscula = valorObtenido.toUpperCase()
   
  
   //otra opcion sin la necesidad de crear una const arreglo
   //if (categories.includes(valorObtenido)){return}
   
   setCategories((val) => {

    const pasadoArregloMayuscula = val.map(e => e.toUpperCase())

    if (pasadoArregloMayuscula.includes(valorObtenidoMayuscula)) {
      return val
    }
      return [valorObtenido,...val]
   })
  }




  return (
    <>
    {/*titulo*/}
    <h1>GifExpertApp</h1> 

    {/*buscardor*/}
      < AddCategory
       //setCategories={setCategories}
       //la utilizo a la funcion para ir a capturar el valor y trabajarlo en el componente padre
       onNewCategory={(valor) => {onAddCategory(valor)} }
        />

      
      {/*gifContainer*/}

      
        {categories.map((categories)=>

        <Gifexp 
        category={categories} 
        key={categories}/>
       )
       }
      

    {/*gif*/}
    </>
  )
}
