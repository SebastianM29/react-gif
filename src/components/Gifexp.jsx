
import { useFetchGifs } from "../hook/useFetchGifs"
import {  GetItem } from "../components"
import  PropTypes from "prop-types";









export const Gifexp = ({ category }) => {
  
  const {isLoading, imagenes } = useFetchGifs(category)

  


 console.log('desde gifexp para el .map',imagenes,isLoading)
  

  return (
    <>
   

    <h3>{ category }</h3>
    {
      isLoading && ( <h2>cargando</h2> )
    }
    <div className="card-grid">
      
      {
        imagenes.map((valor)=>{
         
        return < GetItem 
         key={valor.id}
         {...valor}
         />
        })
      }
    
    
    
    </div>
    

      
    
    </>

  )
}

Gifexp.propTypes = {
  category : PropTypes.string.isRequired
}
