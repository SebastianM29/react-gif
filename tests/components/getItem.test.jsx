import { render, screen } from "@testing-library/react";
import { GetItem } from "../../src/components";

describe('debo obtener los 3 argumentos', () => { 

    const title = 'imagen dragon ball'
    const image = 'http://localhost/imagen'
    const id = '123456'
    

   

    test('snapshot', () => { 
        const{container,getByText} = render(<GetItem title={title} image={image} id={id}/>)
        expect(container).toMatchSnapshot()
    })

    test('recibiendo imagen con el url y el alt alternativo', () => { 

        
         render(<GetItem title={title} image={image} id={id}/>)
        screen.debug()
        //screen.getByRole('img').src o screen.getByRole('img').alt  se desestructura
        const {src,alt} = screen.getByRole('img')
        console.log(src,alt)
        expect(src).toBe(image)
        expect(alt).toBe(title)
      


     })
     test('titulo en parrafo', () => { 

        const {container} = render(<GetItem title={title} image={image} id={id}/>)
        const p = container.querySelector('p');
        expect(p.innerHTML).toContain(title)
        //otra manera de hacerlo 
        //expect (screen.getByText(title)).toBeTruthy()
      })
   
   

 })