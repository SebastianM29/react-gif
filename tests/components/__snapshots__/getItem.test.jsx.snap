// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`debo obtener los 3 argumentos snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="imagen dragon ball"
      src="http://localhost/imagen"
    />
    <p>
      imagen dragon ball
    </p>
  </div>
</div>
`;
